### ISSUE TITLE
Consolekit support

### ISSUE THREAD

**Issue by vberger** [Smithay/smithay#95]
As an alternative session backend to logind, we need to implement Consolekit support.

Apparently, the consolekit API is fully on dbus: https://consolekit2.github.io/ConsoleKit2/

There are apparently a lot of similarities with the logind backend.

**Comment by Drakulix** [comment]
Is this superseded by #127 or a separate protocol, that we still want to support?
Otherwise we could close this issue in favor of #127.

**Comment by vberger** [comment]
Consolekit is a still a different protocol than (e)logind as far as I can tell, so this is not superseded by #127.

Now do we actually want to support ConsoleKit, that is the main question.

Pros: ConsoleKit is available on at least some of the *BSD, while (e)logind is not
Cons: ConsoleKit seems to be unmaintained for years now

**Comment by kchibisov** [comment]
Well, consolekit is being dropped even by gentoo https://www.gentoo.org/support/news-items/2020-04-14-elogind-default.html . Also, I'd say that the primary BSD we should look at is FreeBSD, since a lot of Wayland libs assume linux and this is likely the only BSD that patch a lot of things in it and try to make it work IIRC.

I'm not familiar with what FreeBSD is using though.

**Comment by raspher** [comment]
- https://freedesktop.org/wiki/Software/ConsoleKit/ - unmaintained
- https://github.com/ConsoleKit2/ConsoleKit2 - depends on x11, seems rarely maintained (code frequency) and uses a lot deprecated dependencies
- https://github.com/elogind/elogind - well maintained

just my 2 cents
